test/pkg.jl

```diff
@@ -398,4 +398,73 @@ temp_pkg_dir() do
         @test contains(msg, "$package is not an installed package")
         @test !contains(msg, "signal (15)")
     end
+
+    # partial Pkg.update
+    begin
+        nothingtodomsg = "INFO: No packages to install, update or remove\n"
+
+        ret, out, err = @grab_outputs begin
+            Pkg.rm("Example")
+            Pkg.add("Example")
+        end
+        @test ret == nothing && out == ""
+        @test contains(err, "INFO: Installing Example v")
+
+        ret, out, err = @grab_outputs Pkg.update("Example")
+        @test ret == nothing && out == ""
+        @test contains(err, nothingtodomsg)
+
+        ret, out, err = @grab_outputs begin
+            Pkg.rm("Example")
+            Pkg.add("Example", v"0", v"0.4.1-")
```

Comment:
Yes, it's the same format as the REQUIRE file. It's documented [here](https://github.com/JuliaLang/julia/blob/38c803d2252736612878ccf5b040fb35c4bfa516/base/pkg/pkg.jl#L94-L98), but the docstring is in fact a bit cryptic, and it should probably also be made explicit in the manual page.

In any case, I'll add a comment here.